### ISSUE TITLE
Module parse failed

### ISSUE THREAD

**Issue by outlawcode** [WalletConnect/walletconnect-monorepo#3061]
**Describe the bug**

I'm using reactjs with webpack and babel. Got this error prev week. waste me a week and still can not resolved.

`class-map.js:6 Uncaught (in promise) ReferenceError: t is not defined
    at i.update (class-map.js:6:855)
    at i._$AS (directive.js:6:239)
    at S (lit-html.js:6:2877)
    at k._$AI (lit-html.js:6:6044)
    at M.v (lit-html.js:6:3765)
    at R.g (lit-html.js:6:4955)
    at R._$AI (lit-html.js:6:4392)
    at D (lit-html.js:6:7889)
    at Et.update (lit-element.js:6:407)
    at Et.performUpdate (reactive-element.js:6:4838)`

import {Web3Modal} from "@web3modal/standalone";
import {SignClient} from "@walletconnect/sign-client";

**SDK Version (if relevant)**
 - Client: [e.g. JS, Swift, Kotlin]

`"@babel/cli": "^7.0.0-bridge.0",
		"@babel/compat-data": "7.19.4",
		"@babel/core": "7.22.5",
		"@babel/generator": "7.19.6",
		"@babel/helper-compilation-targets": "^7.22.6",
		"@babel/helper-create-class-features-plugin": "^7.22.6",
		"@babel/helper-module-transforms": "^7.22.5",
		"@babel/plugin-proposal-class-properties": "7.4.0",
		"@babel/plugin-proposal-nullish-coalescing-operator": "^7.18.6",
		"@babel/plugin-proposal-optional-chaining": "^7.21.0",
		"@babel/plugin-syntax-dynamic-import": "7.2.0",
		"@babel/plugin-transform-runtime": "^7.22.5",
		"@babel/polyfill": "7.4.3",
		"@babel/preset-env": "^7.18.2",
		"@babel/preset-react": "^7.17.12",
		"@babel/register": "7.4.0",
		"@babel/runtime": "^7.22.5",
		"add-asset-html-webpack-plugin": "3.1.3",
		"babel-core": "7.0.0-bridge.0",
		"babel-eslint": "10.0.1",
		"babel-loader": "8.0.5",
		"babel-plugin-dynamic-import-node": "2.2.0",
		"babel-plugin-lodash": "3.3.4",
		"babel-plugin-react-intl": "3.0.1",
		"babel-plugin-styled-components": "1.10.0",
		"babel-plugin-transform-react-remove-prop-types": "0.4.24",`

`module: {
        rules: [
			{
				test: /\.(js|jsx|es|cjs)$/, // Transform all .js and .jsx files required somewhere with Babel
				exclude: /node_modules/,
				use: {
					loader: 'babel-loader',
					options: {
						presets: ['@babel/preset-env', '@babel/preset-react'],
					},
				},
			},
			{ test: /node_modules[\\/]@walletconnect/, loader: 'babel-loader' },
			{ test: /node_modules[\\/]@web3modal/, loader: 'babel-loader' },`

**To Reproduce**

**Expected behavior**

Modal should be showing, but not.

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

**Comment by DarshanDixit05** [comment]
Hi 👋 if this issue is open , can i work on it ?

**Comment by dextfprotocol** [comment]
We have the same error with this import:

`import { EthereumProvider } from '@walletconnect/ethereum-provider'`

And the error is

```
ERROR  Failed to compile with 1 error                                                                                                                                                                            
error  in ./node_modules/@walletconnect/ethereum-provider/dist/index.es.js
Module parse failed: Unexpected token (1:1502)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
```

Having formatted (for readability) the `./node_modules/@walletconnect/ethereum-provider/dist/index.es.js` file, the offending code appears to be this:

```
|       rpcMap: p({}, t.length ? { [g(t)]: c[g(t)] } : {})
|     },
>     r = e?.filter((l) => !m.includes(l)),
|     d = i?.filter((l) => !u.includes(l))
|   if (!s && !h && !n && !(r != null && r.length) && !(d != null && d.length))
```

We use Webpack 4.46.0 with Babel.

**Comment by outlawcode** [comment]
> We have the same error with this import:
> 
> `import { EthereumProvider } from '@walletconnect/ethereum-provider'`
> 
> And the error is
> 
> ```
> ERROR  Failed to compile with 1 error                                                                                                                                                                            
> error  in ./node_modules/@walletconnect/ethereum-provider/dist/index.es.js
> Module parse failed: Unexpected token (1:1502)
> You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
> ```
> 
> Having formatted (for readability) the `./node_modules/@walletconnect/ethereum-provider/dist/index.es.js` file, the offending code appears to be this:
> 
> ```
> |       rpcMap: p({}, t.length ? { [g(t)]: c[g(t)] } : {})
> |     },
> >     r = e?.filter((l) => !m.includes(l)),
> |     d = i?.filter((l) => !u.includes(l))
> |   if (!s && !h && !n && !(r != null && r.length) && !(d != null && d.length))
> ```
> 
> We use Webpack 4.46.0 with Babel.

I stuck with same error for 1 week. Still not resolved yet.

**Comment by yuanxxxxxx** [comment]
@outlawcode I also encountered this problem, did you solve it?

**Comment by dextfprotocol** [comment]
For me the problem was with `webpack 4` being unable to deal with some **es2020** characteristics like optional chaining.

I had to upgrade `webpack` from `4.46.0` to `5.88.2` (the latest).

This solved the problem with the “Module parse failed”.

Because `webpack 5` no longer polyfills Node.js core modules automatically, I had to install some compatible modules from npm and use fallbacks (refer to [this documentation](https://webpack.js.org/configuration/resolve/#resolvefallback)). As reference, this is what I added to my `webpack` config:

```
module.exports = {
  //...
  resolve: {
    fallback: {
      assert: require.resolve('assert/'),
      stream: require.resolve('stream-browserify'),
      http: require.resolve('stream-http'),
      https: require.resolve('https-browserify'),
      os: require.resolve('os-browserify/browser')
    },
  },
};
```

Of course I added the required `npm` packages to the `package.json`.
